src/main/java/com/salesforce/storm/spout/dynamic/retry/DefaultRetryManager.java

```diff
@@ -121,7 +121,7 @@ public void failed(MessageId messageId) {
 
         // Determine when we should retry this msg next
         // Calculate how many milliseconds to wait until the next retry
-        long additionalTime = (long) (failCount * getInitialRetryDelayMs() * getRetryDelayMultiplier());
+        long additionalTime = (long) (getInitialRetryDelayMs() * Math.pow(getRetryDelayMultiplier(), failCount));
```

Comment:
We should remove the + 1 when we initialize failCount so that it starts at 0.  Otherwise we never use the initial delay, but skip directly to what should be the delay for the second retry.

Reply:
It makes more sense to me to leave the failCount initialization as is. Otherwise, it would be off by one as far as the true number of times a message has failed, and there is other logic scattered about based on it. Instead, we should just subtract 1 here when calculating the delay.

Reply:
I agree with @mmoldavan here though I totally get where you’re coming from @noshirpatel-salesforce . I’d prefer we leave it as Matt suggests.

One thing I noticed though, for some reason we don’t supply the 1 in getOrDefault() here https://github.com/mmoldavan/storm-dynamic-spout/blob/4c6081fe2570f1576b3f15e47878d8326e6e0b7d/src/main/java/com/salesforce/storm/spout/dynamic/retry/DefaultRetryManager.java#L119 which seems strange. This line also could stand to ha e a comment stating basically what you said above @mmoldavan .

Reply:
Yeah, I see.  That is the increment.  Works for me.